File: vseros/2014-15/okr/coocies_clicker.cpp
#include <iostream>

using namespace std;

int main()
{
	int c, p, n;
	cin >> c >> p >> n;
	int factories = 0;
	int cookie = 0;
	int production = 1;
	int time = 0;
	while (cookie < n)
	{
		int justfinish = (n - cookie + production - 1) / production;
		int nextfactory = (c - cookie + production - 1) / production;
		int hardfinish = (n - (cookie + nextfactory * production - c) + production + p - 1) / (production + p);
		if (nextfactory + hardfinish < justfinish)
		{
			++factories;
			time += nextfactory;
			cookie += nextfactory * production - c;
			production += p;
		}
		else
		{
			time += justfinish;
			cookie += justfinish * production;
		}
	}
	cout << time << endl;
	return 0;
}

File: vseros/2014-15/okr/coocies_clicker.py
c = int(input())
p = int(input())
n = int(input())

cookie = 0
prod = 1
time = 0

without_fact = (n - cookie + prod - 1) // prod
with_fact = (c - cookie + prod - 1) // prod + (n - cookie - ((c - cookie + prod - 1) // prod * prod - c) + prod + p - 1) // (prod + p)
while with_fact < without_fact:
    time += (c - cookie + prod - 1) // prod
    cookie += (c - cookie + prod - 1) // prod * prod - c
    prod += p
    without_fact = (n - cookie + prod - 1) // prod
    with_fact = (c - cookie + prod - 1) // prod + (n - cookie - ((c - cookie + prod - 1) // prod * prod - c) + prod + p - 1) // (prod + p)

time += (n - cookie + prod - 1) // prod
print(time)
